test/common/http/conn_manager_impl_test.cc

```diff
@@ -6049,6 +6050,35 @@ TEST_F(HttpConnectionManagerImplTest, NewConnection) {
   EXPECT_EQ(1U, stats_.named_.downstream_cx_http3_active_.value());
 }
 
+TEST_F(HttpConnectionManagerImplTest, UpstreamHeadersSize) {
+  setup(false, "");
+  EXPECT_CALL(*codec_, dispatch(_)).WillOnce(Invoke([&](Buffer::Instance&) -> Http::Status {
+    RequestDecoder* decoder = &conn_manager_->newStream(response_encoder_);
+    RequestHeaderMapPtr headers{
+        new TestRequestHeaderMapImpl{{":authority", "host"}, {":path", "/"}, {":method", "GET"}}};
+    decoder->decodeHeaders(std::move(headers), true);
+    return Http::okStatus();
+  }));
+
+  setupFilterChain(1, 0);
+
+  std::shared_ptr<Upstream::MockClusterInfo> cluster_info{
+      new NiceMock<Upstream::MockClusterInfo>()};
+
+  EXPECT_CALL(*decoder_filters_[0]->callbacks_, clusterInfo())
```

Comment:
![Screenshot from 2020-06-19 23-16-20](https://user-images.githubusercontent.com/17480353/85197376-38a50e80-b2fe-11ea-86b2-c6abc15f2106.png)
Getting an error here. How to add clusterinfo to callbacks?

Reply:
I think `decoder_filters_[0]` is a  `MockStreamDecoderFilter` which should ideally have set clusterinfo [callback](https://github.com/envoyproxy/envoy/blob/74290ef76a76fbbf50f072dc33438791f93f68c7/test/mocks/http/mocks.cc#L43).